touchforms/formplayer/static/formplayer/script/entrycontrols_full.js

```diff
@@ -1,785 +1,584 @@
     }
 
-    this.lat = lat;
-    this.lon = lon;
-
-    this.$lat.text(lat != null ? (lat >= 0. ? 'N' : 'S') + intpad(Math.abs(lat).toFixed(5), 8) : '??.?????');
-    this.$lon.text(lat != null ? (lon >= 0. ? 'E' : 'W') + intpad(Math.abs(lon).toFixed(5), 9) : '???.?????');
-
-    return true;
-  }
-
-  this.delay_action = function(tag, dofunc, delay) {
-    var timer = this.timers[tag];
-    if (timer) {
-      clearTimeout(timer);
-      timer = null;
+    this.search = function(query) {
+        var map = this.map;
+        this.geocoder.geocode({
+            'address': query
+        }, function(results, status) {
+            if (status == google.maps.GeocoderStatus.OK) {
+                map.fitBounds(results[0].geometry.viewport);
+                map.setCenter(results[0].geometry.location);
+            }
+        });
     }
-    this.timers[tag] = setTimeout(dofunc, 1000 * delay);
-  }
-
-  this.search = function(query) {
-    var map = this.map;
-    this.geocoder.geocode({'address': query}, function(results, status) {
-      if (status == google.maps.GeocoderStatus.OK) {
-        map.fitBounds(results[0].geometry.viewport);
-        map.setCenter(results[0].geometry.location);
-      }
-    });
-  }
 }
+GeoPointEntry.prototype = Object.create(Entry.prototype);
+GeoPointEntry.prototype.constructor = Entry;
 
-function renderQuestion (q, $container, init_answer) {
-  var control = null;
-
-  q.domain_meta = q.domain_meta || {};
-
-  if (q.customlayout != null) {
-    control = q.customlayout();
-  } else if (q.datatype == "str") {
-    if (q.style["raw"] == 'numeric') {
-      control = new PhoneEntry();
-    } else {
-      control = new FreeTextEntry({domain: q.domain, prose: q.domain_meta.longtext});
-    }
-  } else if (q.datatype == "int") {
-    control = new IntEntry();
-  } else if (q.datatype == "longint") {
-    control = new IntEntry(null, 15);
-  } else if (q.datatype == "float") {
-    control = new FloatEntry();
-  //  } else if (q.datatype == "passwd") {
-  //control = new PasswordEntry({domain: q.domain});
-  } else if (q.datatype == "select") {
-      control = new SingleSelectEntry({choices: q.choices, choicevals: q.choicevals, meta: q.domain_meta});
-  } else if (q.datatype == "multiselect") {
-    control = new MultiSelectEntry({choices: q.choices, choicevals: q.choicevals, meta: q.domain_meta});
-  } else if (q.datatype == "date") {
-    control = new DateEntry(q.domain_meta);
-  } else if (q.datatype == "time") {
-    control = new TimeOfDayEntry();
-  } else if (q.datatype == "geo") {
-    control = new GeoPointEntry();
-  } else {
-    control = new UnsupportedEntry(q.datatype);
-  }
-
-  if (control == null) {
-    console.log('no active control');
-    return null;
-  }
-
-  control.setAnswer(init_answer);
-  control.load(q, $container);
-  return control;
-}
+/**
+ * Gets the entry based on the datatype of the Question
+ * @param {Object} question - A Question object
+ */
+function getEntry(question) {
```

Comment:
an `Entry` was referred to as a `Control` `Widget` and `Entry` in the code before. now it will all be `Entry`